main_test.go

```diff
@@ -0,0 +1,215 @@
+							Value: []byte("super-secret-password"),
+						},
+					},
+				},
+			},
+		}
+
+		opts.InjectRequestHeaders = append([]options.Header{authHeader}, opts.InjectRequestHeaders...)
+		opts.InjectResponseHeaders = append(opts.InjectResponseHeaders, authHeader)
+		return opts
+	}
+
+	type loadConfigurationTableInput struct {
+		configContent      string
+		alphaConfigContent string
+		args               []string
+		extraFlags         func() *pflag.FlagSet
+		expectedOptions    func() *options.Options
+		expectedErr        error
+	}
+
+	DescribeTable("LoadConfiguration",
+		func(in loadConfigurationTableInput) {
+			var configFileName, alphaConfigFileName string
+
+			defer func() {
+				if configFileName != "" {
+					Expect(os.Remove(configFileName)).To(Succeed())
+				}
+				if alphaConfigFileName != "" {
+					Expect(os.Remove(alphaConfigFileName)).To(Succeed())
+				}
+			}()
+
+			if in.configContent != "" {
+				By("Writing the config to a temporary file", func() {
+					file, err := ioutil.TempFile("", "oauth2-proxy-test-config-XXXX.cfg")
+					Expect(err).ToNot(HaveOccurred())
+					defer file.Close()
+
+					configFileName = file.Name()
+
+					_, err = file.WriteString(in.configContent)
+					Expect(err).ToNot(HaveOccurred())
+				})
+			}
+
+			if in.alphaConfigContent != "" {
+				By("Writing the config to a temporary file", func() {
+					file, err := ioutil.TempFile("", "oauth2-proxy-test-alpha-config-XXXX.yaml")
+					Expect(err).ToNot(HaveOccurred())
+					defer file.Close()
+
+					alphaConfigFileName = file.Name()
+
+					_, err = file.WriteString(in.alphaConfigContent)
+					Expect(err).ToNot(HaveOccurred())
+				})
+			}
+
+			extraFlags := pflag.NewFlagSet("test-flagset", pflag.ExitOnError)
+			if in.extraFlags != nil {
+				extraFlags = in.extraFlags()
+			}
+
+			opts, err := loadConfiguration(configFileName, alphaConfigFileName, extraFlags, in.args)
+			if in.expectedErr != nil {
+				Expect(err).To(MatchError(in.expectedErr.Error()))
+			} else {
+				Expect(err).ToNot(HaveOccurred())
+			}
+			Expect(in.expectedOptions).ToNot(BeNil())
+			Expect(opts).To(Equal(in.expectedOptions()))
+		},
+		Entry("with legacy configuration", loadConfigurationTableInput{
+			configContent:   testCoreConfig + testLegacyConfig,
+			expectedOptions: testExpectedOptions,
+		}),
+		Entry("with alpha configuration", loadConfigurationTableInput{
+			configContent:      testCoreConfig,
+			alphaConfigContent: testAlphaConfig,
+			expectedOptions:    testExpectedOptions,
+		}),
+		Entry("with bad legacy configuration", loadConfigurationTableInput{
+			configContent:   testCoreConfig + "unknown_field=\"something\"",
+			expectedOptions: func() *options.Options { return nil },
+			expectedErr:     errors.New("failed to load config: error unmarshalling config: 1 error(s) decoding:\n\n* '' has invalid keys: unknown_field"),
+		}),
+		Entry("with bad alpha configuration", loadConfigurationTableInput{
```

Comment:
This looks like just a test case for invalid YAML. Do you have one where the YAML is valid but the underlying schema doesn't match what we expect? (e.g. array where string is expected)

Or do you have more rigorous test cases handled already in the underlying alpha config packages -- making it redundant here?

Reply:
The intention here was just to have basic tests to show we can load yaml and if there's an error, then an error is thrown. Having all of the corner cases tested should be the responsibility of the unit tests for the `LoadYAML` method that this calls IMO, if I haven't got something there (which I don't think I have) I will add it